src/guidellm/utils/dict.py

```
@@ -0,0 +1,23 @@
+def recursive_key_update(d, key_update_func):
+  if not isinstance(d, dict) and not isinstance(d, list):
+    return d
+
+  if isinstance(d, list):
+    for item in d:
+      recursive_key_update(item, key_update_func)
+    return d
+
+  updated_key_pairs = []
+  for key, _ in d.items():
+    updated_key = key_update_func(key)
+    if key != updated_key:
+      updated_key_pairs.append((key, updated_key ))
+
+  for key_pair in updated_key_pairs:
+    old_key, updated_key = key_pair
+    d[updated_key] = d[old_key]
```

Small nit. Might want to throw an exception here in the case that `updated_key` already exists. Could save a bunch of debugging time later.

---

Isn't this handled by line 13 where we prevent any addition to updated_key_pairs if the old key and updated key are the same? So it is impossible for updated_key_pairs to contain anything but updated keys I would say.